controllers/operators/operators_reconciler.go

```diff
@@ -0,0 +1,202 @@
+package operators
+
+import (
+	"context"
+	"fmt"
+
+	sv1b2 "github.com/RedHatInsights/strimzi-client-go/apis/kafka.strimzi.io/v1beta2" //TODO replace ViaQ repo by grafana/loki/operator one when released
+	"github.com/go-logr/logr"
+	gv1alpha1 "github.com/grafana-operator/grafana-operator/v4/api/integreatly/v1alpha1"
+	lv1beta1 "github.com/grafana/loki/operator/api/v1beta1"
+	flowsv1alpha1 "github.com/netobserv/network-observability-operator/api/v1alpha1"
+	"github.com/netobserv/network-observability-operator/controllers/constants"
+	"github.com/netobserv/network-observability-operator/controllers/reconcilers"
+	ov1 "github.com/operator-framework/api/pkg/operators/v1"
+	ov1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	pv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type Reconciler struct {
+	ns          string
+	client      reconcilers.ClientHelper
+	log         logr.Logger
+	desiredSpec *flowsv1alpha1.FlowCollectorSpec
+}
+
+func NewReconciler(ctx context.Context, client reconcilers.ClientHelper, namespace string, spec *flowsv1alpha1.FlowCollectorSpec) *Reconciler {
+	rlog := log.FromContext(ctx, "component", "Operators")
+
+	return &Reconciler{
+		client:      client,
+		ns:          namespace,
+		desiredSpec: spec,
+		log:         rlog,
+	}
+}
+
+// Reconcile the required installed operators
+func (r *Reconciler) Reconcile(ctx context.Context, target *flowsv1alpha1.FlowCollector) error {
+	operators := []string{}
+
+	// install operators if main instance is required
+	// dependent objects will be ignored if InstanceSpec is nil
+	if target.Spec.Loki.InstanceSpec != nil {
```

Comment:
Same comment than here https://github.com/netobserv/network-observability-operator/pull/78#discussion_r861599886 :-)
We need an explicit flag, due to restrictions when the CR is created from a console form (not yaml). Checking nil pointers won't work (they'll never be nil).

By the way, I haven't checked in deep if the Kafka config is consistent with what @OlivierCazade is doing here: https://github.com/netobserv/network-observability-operator/pull/78 , do you know? Things like TopicName will be defined in just one place, right?

Reply:
There is not much difference between the configs. I'll adapt after https://github.com/netobserv/network-observability-operator/pull/78 merge